base_interface.go

```diff
@@ -55,6 +61,38 @@ func requestNew(options options) rest.Request {
 	}
 }
 
+func ExtractEndpoint(link string) (string, error) {
+	parsedURL, err := url.Parse(link)
+	if err != nil {
+		return "", err
+	}
+
+	return parsedURL.Path, nil
+}
+
+func SetHost(request rest.Request, host string) (rest.Request, error) {
+	endpoint, err := ExtractEndpoint(request.BaseURL)
+	if err != nil {
+		return request, err
+	}
+
+	request.BaseURL = host + endpoint
+	return request, nil
+}
+
+func SetDataResidency(request rest.Request, region string) (rest.Request, error) {
```

Comment:
we should add comments to document what these function does, especially how it manipulates the `host` field